File: Sorting_Programs/Quicksort.c
#include<stdio.h>
int partition(int arr[],int start,int end)
{
    int key = arr[start];
    int i = start +1;
    int j = end;
    int temp;
 do{
    while(arr[i]<=key)
    {
        i++;
    }
    while(arr[j]>key)
    {
        j--;
    }
    if(i<j)
    {
       temp = arr[i];
       arr[i]=arr[j];
       arr[j]=temp;
    }

    }while (i<j);

    temp = arr[start];
    arr[start]=arr[j];
    arr[j]=temp;

   return j;
    

}
void QuickSort(int arr[],int start,int end)
{
    if(start<end)
    {     
      int index=partition(arr,start,end);//  1 4 2 6 3 7
      QuickSort(arr,start,index-1);
      QuickSort(arr,index+1,end);
    }
}
int  main()
{
   int arr[100];
    int n;

    printf("Enter the number of elements: ");
    scanf("%d",&n);
    printf("\n");
    printf("Enter the element: ");

    for(int i=0;i<n;i++)
    {
        scanf("%d",&arr[i]);
    }

    printf("The sorted array is: ");
    printf("\n");

    QuickSort(arr,0,n-1);

    for(int i=0;i<n;i++)
    {
       printf("%d ",arr[i]);
    }
    return 0;
}

File: Sorting_Programs/python/quicksort.py
def partition(nums,start,end):
    pivot = nums[end]
    i = start-1
    for j in range(start,end+1):
        if pivot > nums[j]:
            i += 1
            nums[j],nums[i] = nums[i],nums[j]
    i += 1
    nums[i],nums[end] = nums[end],nums[i]
    return i

def quicksort(nums,start,end):
    if end < start:
        return
    index = partition(nums,start,end)
    quicksort(nums,start,index-1)
    quicksort(nums,index+1,end)

nums  = [4,2,7,6,1,0,5,3]

quicksort(nums,0,len(nums)-1)

print(nums)